doc: replace PuTTY with Serial Terminal (#13663)
nrfconnect/sdk-nrf: doc/nrf/test_and_optimize.rst

.. _test_and_optimize:
.. _gs_testing:
.. _serial_terminal_connect:
.. _testing_rtt:
.. _testing_rtt_connect:
.. _putty:
.. _testing_vscode:

Testing and optimization
########################

> |application_sample_definition|
> 
> The :ref:`applications` and :ref:`samples` in the |NCS| come with their own testing instructions.
> Follow these instructions to make sure that the application runs as expected.

Reviewer: I noticed the include is causing a doubling up on links:
![image](https://github.com/nrfconnect/sdk-nrf/assets/83582261/24ebefeb-0868-4e48-9be4-15e8913489a7)

```suggestion
|application_sample_definition|

Each application in the |NCS| come with its own testing instructions.
Follow these instructions to make sure that the application runs as expected.
```

Testing and optimization
########################

Each :ref:`applications` and :ref:`samples` in the |NCS| come with their own testing instructions.
Follow these instructions to make sure that the application runs as expected.

Information about the current state of the application is usually provided through the LEDs or through :term:`Universal Asynchronous Receiver/Transmitter (UART)`, or through both.
See the user interface section of the application's documentation for description of the LED states or available UART commands.

To see the UART output, connect to the development kit (DK) with a terminal emulator.
The default serial port settings are the following:

.. list-table:: Default serial port connection settings for Nordic Semiconductor devices
   :widths: auto
   :header-rows: 1

   * - Baud rate
     - Data bits
     - Stop bits
     - Parity
     - :term:`UART Hardware Flow Control (UART HWFC)`
   * - 115200
     - 8
     - 1
     - No parity
     - None

Use one of the following methods:

.. tabs::

   .. group-tab:: nRF Connect Serial Terminal

      You can use the `nRF Connect Serial Terminal`_ app, which you can install from `nRF Connect for Desktop`_.

>       The nRF Connect Serial Terminal app establishes serial communication with the device, sends commands through UART, and also displays the UART output.

Reviewer: ```suggestion
      It establishes serial communication with the device, sends commands through UART, and also displays the UART output.
```

>       The application is available for Windows, Linux, and macOS.

Reviewer: ```suggestion
      The app is available for Windows, Linux, and macOS.
```
Since nothing new has been introduced, "application" in this context would be referring to the sample or application from the NCS, not Serial Terminal.

      To connect to DK with Serial Terminal, complete the following steps:

      1. Launch the Serial Terminal app.
      #. Connect the DK to the PC with a USB cable.
      #. Power on the DK.
      #. Click :guilabel:`Select Device` and select the particular kit entry from the drop-down list in the Serial Terminal.
      #. Select the serial port you want to connect to from the drop-down list.

>       #. Configure the serial port connection parameters.
>          See the `Serial Terminal configuration`_ page for more information.

Reviewer: ```suggestion
      #. Configure the default serial port connection settings.

         See the `Serial Terminal configuration`_ page of the Serial Terminal documentation for more information.
```
I am not sure how to word the first line, but it should point to the default values introduced earlier on the page.

Not sure how useful the link is. It provides information on what each setting means, when I would have expected more information on how to set those.

This step is also additional information. Neither the old version or the Getting started page for Serial Terminal mention needing to configure the settings before connecting. Is it necessary here?

      #. Click :guilabel:`Connect to port`.

      nRF Connect Serial Terminal starts the serial communication with the kit.

>       For more information about how to connect using the app, see the `Serial Terminal documentation <Connecting using Serial Terminal_>`_.

Reviewer: I was expecting the link to take me to the front page of the documentation, the target page isn't really wrong either.

Author: Yeah, something is a bit off. Adjusted the link phrasing.

   .. group-tab:: nRF Connect for Visual Studio Code

      The |nRFVSC| includes an integrated serial port and RTT terminal, which you can use to connect to your board.
      For detailed instructions, see `How to connect to the terminal`_ on the `nRF Connect for Visual Studio Code`_ documentation site.

      The extension is available for Windows, Linux, and macOS.

>    .. group-tab:: PuTTY (Windows only)

Reviewer: Does "Windows only" need to be in the tab name? It looks off to me.

>       PuTTY is available only for Windows.

Reviewer: I would like some sort of one-sentence description of PuTTY, to match the previous two tabs.

>       To connect to DK with PuTTY using CDC-UART, complete the following steps:

Reviewer: ```suggestion
      To connect to the DK with PuTTY using CDC-UART, complete the following steps:
```

>       1. Connect the DK to the computer with a USB cable, and then turn on the DK.
> 
>          **LED1** starts blinking.
> 
>       #. Open PuTTY.

Reviewer: ```suggestion
      1. Open the PuTTY application.
      #. Connect the DK to the PC with a USB cable.
      #. Power on the DK.
```
To match Serial Terminal.

         * Implicit CR in every LF
         * Implicit LF in every CR
         * Local echo: Force on
         * Local line editing: Force on

         .. figure:: images/putty.svg
            :alt: PuTTY configuration for sending commands through UART

            PuTTY configuration for sending commands through UART

      #. Click the **Serial** category under the **Connection** category in the category selection tree to see options controlling the local serial line.
      #. Type the COM port corresponding to your DK in the **Serial line to connect to** field.

         Depending on what devices you have connected to your computer, you might have several choices.
         To find the correct port:

         a. Right-click on the Windows Start menu, and select **Device Manager**.
         #. In the **Device Manager** window, scroll down and expand **Ports (COM & LPT)**.
         #. Find the port named *JLink CDC UART Port* and note down the number in parentheses.

            If you have more than one J-Link UART Port, unplug the one that you want to use, plug it back in, and observe which one appeared last.

            Your DK can show up as two consecutive COM ports.

>             If this is the case, you need to test which COM port is the correct one.

Reviewer: Should this information be also in the Serial Terminal section? Or somewhere outside of the tabs? It doesn't make sense to exist only here.

Author: It is already covered to some degree here: https://docs.nordicsemi.com/bundle/nrf-connect-serial-terminal/page/selecting_serial_port.html
I'd rather update this in the ST guide and rely on the link to the ST docs.

Author: Tasked here https://nordicsemi.atlassian.net/browse/NCD-685

>       #. Configure the following settings in the **Configure the serial line** section using the values from the Default serial port connection settings table above.

Reviewer: ```suggestion
      #. Configure the settings in the **Configure the serial line** section using the values from the Default serial port connection settings table above.
```

      #. Click :guilabel:`Open`.

      The terminal window opens.

>       UART can also be used for logging purposes as one of the :ref:`logging backends <ug_logging_backends>`.

Reviewer: Since the tab is specifically about PuTTY, should this line be elsewhere?

   .. group-tab:: RTT

>       To view the logging output using Real Time Transfer (RTT), modify the configuration settings of the application to override the default UART console:

Reviewer: This tab should also have some sort of introduction, and list which OSes are supported to match the other tabs.

>       To view the logging output using Real Time Transfer (RTT), modify the configuration settings of the application to override the default UART console:

Reviewer: ```suggestion
      To view the logging output using :term:`Real-Time Transfer (RTT)`, modify the configuration settings of the application to override the default UART console:
```

      .. code-block:: none

         CONFIG_USE_SEGGER_RTT=y
         CONFIG_RTT_CONSOLE=y
         CONFIG_UART_CONSOLE=n

>       SEGGER's J-Link RTT can also be used for logging purposes as one of the :ref:`logging backends <ug_logging_backends>`.

Reviewer: Should RTT be mentioned as an alternative to UART in the beginning of the page? It's not mentioned at all before this tab.

Author: Moved later on the page.

      .. note::

         SEGGER's J-Link RTT is part of the `J-Link Software and Documentation Pack`_.

>          You must have this software installed on your platform to use RTT.

Reviewer: This seems like it should be part of the introduction for this tab, not a note after a couple of lines.

Author: Reworked the tab to be more about SEGGER's J-Link RTT Viewer.

      To run RTT on your platform, complete the following steps:

      1. From the J-Link installation directory, open the J-Link RTT Viewer application:

         * On Windows, the executable is called :file:`JLinkRTTViewer.exe`.

>          * On Linux, the executable is called :file:`JLinkRTTViewerExe`.

Reviewer: Implies macOS is not supported through exclusion.
